### ISSUE TITLE
`litellm` conflict with Langroid's `pydantic` required version

### ISSUE THREAD

**Issue by nathnx** [langroid/langroid#497]
I encounter an error when trying to use `litellm/gemini/gemini-pro` via `langroid.language_models.openai_gpt.OpenAIGPTConfig`.

```
langroid.exceptions.LangroidImportError: litellm is not installed by default with Langroid.
```

Although I install `litellm` from langroid extras, the error still persisting.  So, I try to `import litellm` in python session to verify if the library can be used in my env. Unfortunately, the following error occured.

```
..../lib/python3.11/site-packages/litellm/proxy/_types.py", line 1, in <module>
    from pydantic import BaseModel, Extra, Field, model_validator, Json, ConfigDict
ImportError: cannot import name 'model_validator' from 'pydantic' 
```
So, it has conflict with `model_validator` which is being used in `litellm` (see library upgrade: https://github.com/BerriAI/litellm/issues/3417). It means Pydantic `1.10.13` required by Langroid `0.1.262` does not compatible with `litellm (>=1.30.1,<2.0.0)`.

**To Reproduce**

Setup the following config and parse to `SQLChatAgent`
```
from langroid.language_models.openai_gpt import OpenAIGPTConfig
from langroid.agent.special.sql.sql_chat_agent import (
    SQLChatAgent,
    SQLChatAgentConfig,
)

llm_config=OpenAIGPTConfig(
                chat_model="litellm/gemini/gemini-pro",
                temperature=0)

agent = SQLChatAgent(
                  config=SQLChatAgentConfig(
                      llm=llm_config, 
                      ...,
        )
    )
```

**Comment by pchalasani** [comment]
Thanks for raising this @nathnx , yes I'm doing a PR to upgrade langroid to pydantic 2, which should fix this.

**Comment by pchalasani** [comment]
@nathnx this is fixed in 0.1.263 (pydantic v2 support #498 )

**Comment by pchalasani** [comment]
Actually it wasn't quite fixed; This is fixed in 0.1.264 which now installs pydantic 2.x by default.
